### PR TITLE
feat: Physics bullet integration

### PR DESCRIPTION
**Pull request by xavi1111**

### CODE REVIEW

File: Source/ModulePhysics.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+ModulePhysics::~ModulePhysics()
+{
+}
+
+bool ModulePhysics::Init()
+{
+	//collision configuration contains default setup for memory, collision setup. Advanced users can create their own configuration.
+	btDefaultCollisionConfiguration* collisionConfiguration = new btDefaultCollisionConfiguration();
```
Comment (gerard-queralt) [inline Source/ModulePhysics.cpp:15]: All the uses of `new` inside this method don't have the accompaning `delete`s, which means the allocated memory is never freed. If we don't need to save this objects as members, can't we just use references?
```suggestion
	btDefaultCollisionConfiguration collisionConfiguration;
```
And if we need to save them, consider using unique pointers.

File: Source/ModulePhysics.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	btSequentialImpulseConstraintSolver* solver = new btSequentialImpulseConstraintSolver;
+
+	btDiscreteDynamicsWorld* dynamicsWorld = new btDiscreteDynamicsWorld(dispatcher, overlappingPairCache, solver, collisionConfiguration);
+
+	dynamicsWorld->setGravity(btVector3(0, -10, 0));
+
+    return true;
+}
+
+update_status ModulePhysics::Update()
```
Comment (gerard-queralt) [inline Source/ModulePhysics.cpp:33]: Can be deleted.

File: Source/ModulePhysics.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+{
+    
+    return update_status::UPDATE_CONTINUE;
+}
+
+
+bool ModulePhysics::CleanUp()
```
Comment (gerard-queralt) [inline Source/ModulePhysics.cpp:40]: Can be deleted.

File: Source/ModulePhysics.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "Module.h"
+
+class ModulePhysics
+{
+public:
+	ModulePhysics();
+	~ModulePhysics();
```
Comment (gerard-queralt) [inline Source/ModulePhysics.h:9]: Missing `override`.
```suggestion
	~ModulePhysics() override;
```

File: Source/ModulePhysics.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+	bool Init();
```
Comment (gerard-queralt) [inline Source/ModulePhysics.h:11]: Missing `override`.
```suggestion
	bool Init() override;
```

**Review by gerard-queralt — CHANGES_REQUESTED** [review]

File: Source/ModulePhysics.cpp (hunk 1 above)
Comment (xavi1111) [reply Source/ModulePhysics.cpp:15]: Yeah I agree, but this code is meerly to check the integration of the library it is not intented to stay as is.  It is the HelloWorld provided by the library and the only reason it is in this PR is to check that it compiles in more than one machine (mine).

Comment (gerard-queralt) [reply Source/ModulePhysics.cpp:15]: Ah whoops, didn't realize this wasn't pointed to `develop`. Carry on then! :P

**Review by OriolSors — APPROVED** [review]

**Review by Dedado3000 — APPROVED** [review]

**Pull request merged**